### DISCUSSION TITLE
Alternative X Server for win-kex (kali-linux)

### DISCUSSION THREAD

**Discussion by dDanissimo** [microsoft/WSL#6223] [Q&A]
Hello everyone! I've searched for a win-kex tracker issue site for a little while, as DHowett offered to a user in #6108, but decided to first address that issue in this repo.

I've had enough BS with using the kex package with the built-in **Vcxsrv X Server**. It generates a lot of errors, I could troubleshoot them for hours.

However, I use **X410** for my WLinux, and in terms of Pengwin, it works _flawlessly_. I managed to kill Vcxsrv on Kali, launch X410, and launch `kex --sl -s` (stuff breaks only with seamless mode, TigerVNC with the windowed works nothing but alright), and it worked OK! No generated errors, smooth performance. I assume Firewall messes with Vcxsrv, there was no prompt to allow access to pulseaudio and the X Server, like in [Bombal's video](https://youtu.be/nXThnFxwH9c?t=446), due to the fact that I totally purged Windows Defender from the system with a tweaker (it's a totally different branch of this issue, including the removal of a registry key in 20H2 by MS previously used to disable Windows Defender, but nonetheless, it may be also relevant). I tried manually adding inbound rules to the firewall, the paths to the executables are correct, but no progress so far. The thing is that supposedly the net traffic should not be filtered because of Defender Service being totally dead with all attached apps (such as Firewall). So this is probably why I don't see this:
![image](https://user-images.githubusercontent.com/56019205/98648550-7c4cf300-2347-11eb-958d-f041bb6aa980.png)

What I've experienced with Vcxsrv and Seamless Mode so far:

- launch non-root, opens X, no visual output
- launch root, generates an error dialog window, then somehow works on (?!)
- try and modify the display variable, etc. ==> no result

To conclude all this, maybe the devs really hardcoded Vcxsrv, and they were expecting to use it exceptionally, I looked at the bash script /usr/bin/kex, seems like you can modify XSERV variable or something like that, you actually have to edit some more lines in the file.

My winver states:
![image](https://user-images.githubusercontent.com/56019205/98648788-cc2bba00-2347-11eb-8637-3cbf3418baaa.png)

If somebody really went this far and edited the bash-file, can you share the edited file, please? Any other suggestions on that topic would be really appreciated. Yeah, windowed mode seems ok to me, but seamless mode look cutting-edge-like stuff to me, never had to deal with anything like that, willing to test it out as soon as possible.

**Comment by hceuterpe** [comment]
Literally comment the line out entirely (adding # at the beginning of the line) that sets the variable at the beginning of the script.  I had to do this myself since I'm using a third party firewall, and it's not liking the wsl2 UNC path.  Plus like you said, X410 is superior to Vcxsrv.  Doing this simply makes Vcxsrv not start since the variable is now not even set.  Also the other lines trying to terminate Vcxsrv aren't necessary either since it was never started in the first place.  I still get audio through pulseaudio.  Command I'm using:  ```kex --sl -s```

```#XSERV="/usr/lib/win-kex/VcXsrv/vcxsrv.exe"```

**Comment by just1a-person** [comment]
Have you tried `kex --esm`?

**Reply by rommelsanthiago** [reply]
Yeah that worked for me - `kex --esm -i -s`
